app/src/main/java/com/sebastianvm/mango/database/MangoDatabase.kt

```diff
@@ -0,0 +1,38 @@
+package com.sebastianvm.mango.database
+
+import android.content.Context
+import androidx.room.Database
+import androidx.room.Room
+import androidx.room.RoomDatabase
+import com.sebastianvm.mango.database.dao.JobDao
+import com.sebastianvm.mango.database.models.JobEntity
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+
+@Database(entities = [JobEntity::class], version = 1)
+abstract class MangoDatabase : RoomDatabase() {
+    abstract fun jobDao(): JobDao
+}
+
+@Module
+@InstallIn(SingletonComponent::class)
+object MangoDatabaseModule {
+
+    @Provides
+    fun provideMangoDatabase(@ApplicationContext appContext: Context): MangoDatabase {
+        return Room.databaseBuilder(
+            appContext,
+            MangoDatabase::class.java,
+            "MangoDatabase"
+        ).build()
+    }
+
+    @Provides
+    fun provideJobDao(mangoDatabase: MangoDatabase): JobDao {
+        return mangoDatabase.jobDao()
+    }
+
+}
```

Comment:
what do you think about 1 database module that exposes all the daos versus 1 module per dao

Idk if we want these scoped at a viewmodel or singleton level though 🤔

Reply:
I'm fine with having a single module.

I think scoping to the ViewModel should be fine? I think this also prevents us from using it in other places than the ViewModel (e.g. activity or fragment). If we see any issues we can always change it later